[Bug]: Server crashes on start - routes/v2/foodRoutes not compiled in latest image

## Environment
- Image: codewithcj/sparkyfitness_server:latest
- Image created: ~2 hours ago (sha: 507bd13435fd)
- Previous working image: 601e02f50345 (5 days old)

## Error
Server crashes immediately on start:
Error: Cannot find module './routes/v2/foodRoutes'

## Root Cause
The file /app/SparkyFitnessServer/routes/v2/foodRoutes.ts exists in the image
but was not compiled to .js. Node.js cannot load TypeScript files directly.

routes/v2/ only contains foodRoutes.ts instead of foodRoutes.js

## Workaround
Pinning to the previous image (601e02f50345) works fine.

> Hey [@TMG-Education](https://github.com/TMG-Education) thanks for reporting this. Sorry about that, I missed a command in the production dockerfile that isn't needed in the dev version. I've made a PR with a fix.

I am still facing this error, and the docker image is already up to date, do i have to do anything else?

Error: Cannot find module './routes/v2/foodRoutes'

--- a/SparkyFitnessFrontend/src/api/Diary/foodEntryService.ts
+++ b/SparkyFitnessFrontend/src/api/Diary/foodEntryService.ts
@@ -2,13 +2,7 @@ import { apiCall } from '../api';
 import type { MealFood } from '@/types/meal';
 import type { FoodEntryMeal } from '@/types/meal';
 import type { FoodEntry } from '@/types/food';
-import { DayData, Goals } from '@/types/diary';
-
-export interface FoodEntryUpdateData {
-  quantity?: number;
-  unit?: string;
-  variant_id?: string | null;
-}
+import { DayData, FoodEntryUpdateData, Goals } from '@/types/diary';
 
 export interface FoodEntryCreateData {
   user_id?: string;

--- a/SparkyFitnessFrontend/src/hooks/Diary/useFoodEntries.ts
+++ b/SparkyFitnessFrontend/src/hooks/Diary/useFoodEntries.ts
@@ -15,7 +15,6 @@ import {
   updateFoodEntryMeal,
   deleteFoodEntryMeal,
   type FoodEntryMealUpdateData,
-  FoodEntryUpdateData,
   loadDiaryGoals,
 } from '@/api/Diary/foodEntryService';
 
@@ -23,6 +22,7 @@ import { goalKeys } from '@/api/keys/goals';
 import { foodEntryKeys, foodEntryMealKeys } from '@/api/keys/diary';
 import i18n from '@/i18n';
 import { useFoodEntryInvalidation } from '../useInvalidateKeys';
+import { FoodEntryUpdateData } from '@/types/diary';
 
 export const useFoodEntries = (date: string) => {
   const { t } = useTranslation();

--- a/SparkyFitnessFrontend/src/pages/Diary/Diary.tsx
+++ b/SparkyFitnessFrontend/src/pages/Diary/Diary.tsx
@@ -469,6 +469,7 @@ const Diary = () => {
           entry={editingEntry}
           open={true}
           onOpenChange={(open) => !open && setEditingEntry(null)}
+          availableMealTypes={availableMealTypes ?? []}
         />
       )}
 

--- a/SparkyFitnessFrontend/src/pages/Diary/EditFoodEntryDialog.tsx
+++ b/SparkyFitnessFrontend/src/pages/Diary/EditFoodEntryDialog.tsx
@@ -30,17 +30,20 @@ import { useUpdateFoodEntryMutation } from '@/hooks/Diary/useFoodEntries';
 import { calculateNutrition } from '@/utils/nutritionCalculations';
 import { NutrientGrid } from './NutrientsGrid';
 import { useUnitConversion } from '@/hooks/Foods/useUnitConversion';
+import { FoodEntryUpdateData, MealTypeDefinition } from '@/types/diary';
 
 interface EditFoodEntryDialogProps {
   entry: FoodEntry | null;
   open: boolean;
   onOpenChange: (open: boolean) => void;
+  availableMealTypes: MealTypeDefinition[];
 }
 
 const EditFoodEntryDialog = ({
   entry,
   open,
   onOpenChange,
+  availableMealTypes,
 }: EditFoodEntryDialogProps) => {
   const { loggingLevel, energyUnit, convertEnergy } = usePreferences();
 
@@ -48,6 +51,7 @@ const EditFoodEntryDialog = ({
   const [selectedVariantId, setSelectedVariantId] = useState<string | null>(
     entry?.variant_id || null
   );
+  const [mealId, setMealId] = useState<string>(entry?.meal_type_id ?? '');
 
   const { data: customNutrients } = useCustomNutrients();
   const { data: foodData, isLoading: isLoadingFood } = useFoodView(
@@ -170,13 +174,15 @@ const EditFoodEntryDialog = ({
           ...convertedVariant,
           ...savedVariant,
         };
+        const data: FoodEntryUpdateData = {
+          quantity,
+          unit: variantWithId.serving_unit,
+          variant_id: variantWithId.id || null,
+          meal_type_id: mealId,
+        };
         await updateFoodEntry({
           id: entry.id,
-          data: {
-            quantity,
-            unit: variantWithId.serving_unit,
-            variant_id: variantWithId.id || null,
-          },
+          data,
         });
         info(
           loggingLevel,
@@ -200,6 +206,7 @@ const EditFoodEntryDialog = ({
       const updateData = {
         quantity,
         unit: selectedVariant.serving_unit,
+        meal_type_id: mealId,
         variant_id:
           selectedVariant.id === 'default-variant' ? null : selectedVariant.id,
       };
@@ -254,7 +261,7 @@ const EditFoodEntryDialog = ({
                 )}
               </div>
 
-              <div className="grid grid-cols-2 gap-4">
+              <div className="grid grid-cols-3 gap-4">
                 <div>
                   <Label htmlFor="quantity">Quantity</Label>
                   <Input
@@ -298,6 +305,21 @@ const EditFoodEntryDialog = ({
                       )}
                       <SelectSeparator />
                       <SelectItem value="__custom__">Custom unit...</SelectItem>
+                    </SelectContent>
+                  </Select>
+                </div>
+                <div>
+                  <Label htmlFor="meal">Meal</Label>
+                  <Select value={mealId} onValueChange={setMealId}>
+                    <SelectTrigger>
+                      <SelectValue />
+                    </SelectTrigger>
+                    <SelectContent>
+                      {availableMealTypes.map((mealType) => (
+                        <SelectItem key={mealType.id} value={mealType.id}>
+                          {mealType.name}
+                        </SelectItem>
+                      ))}
                     </SelectContent>
                   </Select>
                 </div>

--- a/SparkyFitnessFrontend/src/types/diary.ts
+++ b/SparkyFitnessFrontend/src/types/diary.ts
@@ -46,3 +46,10 @@ export interface MealTypeDefinition {
   is_visible?: boolean;
   show_in_quick_log?: boolean;
 }
+
+export interface FoodEntryUpdateData {
+  quantity?: number;
+  unit?: string;
+  variant_id?: string | null;
+  meal_type_id?: string;
+}

--- a/SparkyFitnessFrontend/src/types/food.ts
+++ b/SparkyFitnessFrontend/src/types/food.ts
@@ -74,6 +74,7 @@ export interface FoodEntry {
   meal_id?: string; // New field for aggregated meals - will be deprecated/null for new meal component entries
   food_entry_meal_id?: string; // New field to link to food_entry_meals parent
   meal_type: string;
+  meal_type_id?: string;
   quantity: number;
   unit: string;
   variant_id?: string;

